### PR TITLE
Fix broken data migration

### PR DESCRIPTION
**Pull request by vkrmis**
**Story card:** none
## Because

The Data migration to copy numbers from telemed v1 to telemed v2 was broken

### CODE REVIEW

File: db/data/20200612054906_copy_teleconsultation_phone_numbers_to_json.rb [hunk 1]
```diff
@@ -1,13 +1,18 @@
 class CopyTeleconsultationPhoneNumbersToJson < ActiveRecord::Migration[5.2]
   def up
-    Facility.where(enable_teleconsultation: true).each do |facility|
-      facility.teleconsultation_phone_numbers = [{"isd_code" => facility.teleconsultation_isd_code,
-                                                  "phone_number" => facility.teleconsultation_phone_number}]
+    Facility.all.each do |facility|
```
Comment (harimohanraj89) [inline db/data/20200612054906_copy_teleconsultation_phone_numbers_to_json.rb:3]: What's the bug? We had facilities where teleconsult was disabled, but they had phone numbers?

**Comment by harimohanraj89** [comment]
PS is it possible to re-run this migration correctly in SBX/QA etc, or is that a lost cause now?

File: db/data/20200612054906_copy_teleconsultation_phone_numbers_to_json.rb (hunk 1 above)
Comment (vkrmis) [reply db/data/20200612054906_copy_teleconsultation_phone_numbers_to_json.rb:3]: That's also a possibility, but the bug here was that `teleconsultation_phone_numbers=` used to simply accept some JSON and set it, but now its a defined setter that failed to pick up string keys we're passing in here.

**Comment by vkrmis** [comment]
> PS is it possible to re-run this migration correctly in SBX/QA etc, or is that a lost cause now?

Should be doable, the old columns were untouched.

**Comment by harimohanraj89** [comment]
Neat. If the originals numbers were left untouched, I'd lean heavily towards writing a _new_ migration to do this correctly, and leaving a comment in this one. Editing migrations that have been run in some envs is a slippery slope. For example, if any other migrations were run after this, we'll have to roll them back as well to be able to rerun this correctly.

Shall we extract into a new migration?

**Comment by vkrmis** [comment]
> Neat. If the originals numbers were left untouched, I'd lean heavily towards writing a _new_ migration to do this correctly, and leaving a comment in this one. Editing migrations that have been run in some envs is a slippery slope. For example, if any other migrations were run after this, we'll have to roll them back as well to be able to rerun this correctly.
> 
> Shall we extract into a new migration?

I'd be _okay_ to manually re-run this since its only run in SBX/QA, but generally I'd agree that new migration > edit migration

**Comment by harimohanraj89** [comment]
Yup, it's simple enough. Let's do a new one, avoid the confusion. Makes it super straightforward to deploy 👍

**Comment by vkrmis** [comment]
@harimohanraj89 Wdyt about sticking a `.with_indifferent_access` on each phone number in the 
setters so we don't have this problem?

**Comment by harimohanraj89** [comment]
Love it 😄 I was considering doing that too, I'll let you decide whether you wanna include in this PR or a separate one. Either way works.

File: db/data/20200612054906_copy_teleconsultation_phone_numbers_to_json.rb [hunk 2]
```diff
@@ -1,5 +1,6 @@
 class CopyTeleconsultationPhoneNumbersToJson < ActiveRecord::Migration[5.2]
   def up
+    # This data migration did not work correctly, the `teleconsultation_phone_numbers=` method didn't handle string keys
```
Comment (harimohanraj89) [inline db/data/20200612054906_copy_teleconsultation_phone_numbers_to_json.rb:3]: Neat. Let's add the filename of the working migration to this comment too for good measure.

File: db/data/20200619062218_copy_teleconsultation_phone_numbers_to_json_fixed.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
+class CopyTeleconsultationPhoneNumbersToJsonFixed < ActiveRecord::Migration[5.2]
+  def up
+    Facility.all.each do |facility|
```
Comment (harimohanraj89) [inline db/data/20200619062218_copy_teleconsultation_phone_numbers_to_json_fixed.rb:3]: Consider `#find_each`.

**Pull request merged**